Automattic/jetpack
# AI Client: Handle token fetching errors from inside the SuggestionEventSource

Errors that happen when fetching the JWT were not handled properly, resulting on an invalid state for the application when it happens.

## Proposed changes:

* Move the token fetching logic to the SuggestionEventSource so we can handle exceptions there and dispatch the proper error event when it happens
* Start dispatching the `ERROR_REQUEST` with the `ERROR_NETWORK` details when some problem happens on the JWT fetching operation
* Fix typo on `handleNetworkError` callback name

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor and open your browser console on the network tab
* Insert a Jetpack Form block on the editor
* On the browser console, change the network throttling setting to "Offline" so we simulate a connection problem
* On the AI Extension, ask for some form (for example, "a post-meetup feedback form")
* Before this change, the extension would be stuck on the requesting state, because the JWT request would fail and the error would not be handled
* After this change, you will see an "It was not possible to process your request. Mind trying again?" message and the orange state on the assistant icon, indicating an error:

<img width="600" alt="Screenshot 2023-08-08 at 15 41 29" src="https://github.com/Automattic/jetpack/assets/6760046/739c4abb-607e-4d8b-b860-fc8862e9b859">
